CMakeLists.txt

```diff
@@ -332,12 +332,62 @@ add_custom_command(                       OUTPUT  ${syscall_macros_h}
   DEPENDS ${PROJECT_SOURCE_DIR}/scripts/gen_syscall_header.py
   )
 
-# This command is a hack to support commands that are always run. Any
-# target that depends on always_rebuild will always be rebuilt.
-add_custom_command(OUTPUT always_rebuild COMMAND cmake -E echo Building for board ${BOARD})
 
-set(syscall_list_h ${CMAKE_CURRENT_BINARY_DIR}/include/generated/syscall_list.h)
-set(syscalls_json  ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/syscalls.json)
+set(syscall_list_h   ${CMAKE_CURRENT_BINARY_DIR}/include/generated/syscall_list.h)
+set(syscalls_json    ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/syscalls.json)
+set(syscalls_subdirs ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/syscalls_subdirs.txt)
+set(syscalls_trigger ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/syscalls_subdirs.trigger)
+
+# When running CMake it must be ensured that all dependencies are correctly aquired.
+execute_process(
+  COMMAND
+  ${PYTHON_EXECUTABLE}
+  ${PROJECT_SOURCE_DIR}/scripts/subfolder_list.py
+  --directory        ${PROJECT_SOURCE_DIR}/include # Walk this directory
+  --out-file         ${syscalls_subdirs}           # Write this file
+  )
+file(READ ${syscalls_subdirs} PARSE_SYSCALLS_PATHS_DEPENDS)
+string(REGEX REPLACE "\n" ";" PARSE_SYSCALLS_PATHS_DEPENDS ${PARSE_SYSCALLS_PATHS_DEPENDS})
```

Comment:
I think these two invocations are equivalent to

  file(
    STRINGS
    ${syscalls_subdirs}
    PARSE_SYSCALLS_PATHS_DEPENDS
  )

Reply:
seems you're right.
Will make code cleaner.